internal/mesh/internal/types/proxy_state_template.go

```diff
@@ -61,3 +66,136 @@ func RegisterProxyStateTemplate(r resource.Registry) {
 		},
 	})
 }
+
+func ValidateProxyStateTemplate(res *pbresource.Resource) error {
+	// TODO(v2): validate a lot more of this
+
+	var pst pbmesh.ProxyStateTemplate
+
+	if err := res.Data.UnmarshalTo(&pst); err != nil {
+		return resource.NewErrDataParse(&pst, err)
+	}
+
+	var merr error
+
+	if pst.ProxyState != nil {
```

Comment:
Nit: we can get rid of some nesting by early returning when `pst.ProxyState` is `nil`.

Reply:
I did not do the early return as any additional validation of the 3 sibling fields would immediately need to reverse it.